src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -423,7 +499,10 @@ class TaskExecutor {
         std::chrono::duration_cast<std::chrono::milliseconds>(task_latency)
             .count();
     processing_container->update_throughput(1, task_latency_micros);
-    latency_hist->insert(static_cast<int64_t>(task_latency_millis));
+    if (cur_model_metric) {
+      (*cur_model_metric)
+          .latency_->insert(static_cast<int64_t>(task_latency_millis));
```

Comment:
This should be micros: `._latency_->insert(static_cast<int64_t>(task_latency_micros))`. We can also remove `task_latency_millis` (unused)